File: Julia/Algorithms/gnome_sort.jl
function gnome_sort!(array)
    index = 1

    while index <= length(array)
        if index == 1
            index += 1
        elseif array[index] > array[index-1]
            index += 1
        else
            array[index], array[index-1] = array[index-1], array[index]
            index -= 1
        end
    end
    return array
end

array = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1]

println(gnome_sort!(array))

File: Lua/Algorithms/gnomeSort.lua
function gnomeSort(array)
    local index = 1
    
    while index <= #array do
        if index == 1 then
            index = index + 1
        elseif array[index] >= array[index - 1] then
            index = index + 1
        else
            local temp = array[index]
            array[index] = array[index - 1]
            array[index - 1] = temp
            index = index - 1
        end
    end
    return array
end

array = {10, 9, 8, 7, 6, 5, 4, 3, 2, 1}

print(table.concat(gnomeSort(array), ", "))
